### PR TITLE
Merge grails 6.x upgrade to 7.0

### PR DESCRIPTION
**Pull request by RxL-Farhan-Ali**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,106 +1,185 @@
...
+    testImplementation("org.grails:grails-gorm-testing-support")
+    testImplementation("org.grails:grails-web-testing-support")
+    testImplementation("org.grails.plugins:geb")
+    testImplementation("org.seleniumhq.selenium:selenium-api:4.19.1")
+    testImplementation("org.seleniumhq.selenium:selenium-remote-driver:4.19.1")
+    testImplementation("org.seleniumhq.selenium:selenium-support:4.19.1")
+    testImplementation("org.spockframework:spock-core")
+    testRuntimeOnly("org.seleniumhq.selenium:selenium-chrome-driver:4.19.1")
+    testRuntimeOnly("org.seleniumhq.selenium:selenium-firefox-driver:4.19.1")
+    testRuntimeOnly("org.seleniumhq.selenium:selenium-safari-driver:4.19.1")
```
Comment (parveen-rx) [inline build.gradle]: Remove selenium dependencies as we don't use the same.

File: build.gradle [hunk 2]
```diff
@@ -1,106 +1,185 @@
...
+
+    repositories {
+        maven {
+            name = "NexusRepo"
+            credentials {
+                username project.findProperty("nexusUsername") ?: System.getenv("NEXUS_USERNAME")
+                password project.findProperty("nexusPassword") ?: System.getenv("NEXUS_PASSWORD")
+            }
+            url project.findProperty("nexusUrl") ?: System.getenv("NEXUS_URL")
+            allowInsecureProtocol = true
```
Comment (parveen-rx) [inline build.gradle]: update as per below
allowInsecureProtocol = project.findProperty("isNexusUrlInsecure") ? true : false

File: grails-cli.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+applicationType: web
+defaultPackage: grails.cc.upgrade.sample
```
Comment (parveen-rx) [inline grails-cli.yml]: update default package

File: settings.gradle [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+pluginManagement {
+    repositories {
+        mavenLocal()
+        maven { url "https://repo.grails.org/grails/core/" }
+        gradlePluginPortal()
+    }
+    plugins {
+        id "org.grails.grails-web" version "6.1.2"
+        id "org.grails.grails-gsp" version "6.1.2"
+//        id "com.bertramlabs.asset-pipeline" version "4.3.0"
```
Comment (parveen-rx) [inline settings.gradle]: remove if not required, also check if webdriver binaries are really needed?

File: src/integration-test/groovy/spring/session/demo/app/SessionPluginSpec.groovy [hunk 5]
```diff
@@ -1,73 +1,78 @@
...
-
-        then: "Key [name] should be expired"
-        $().text() == "NO-KEY"
-    }
-
-    def setMutableUpdateTrue(ConfigObject configObject) {
-        grailsApplication.config.springsession = configObject
-    }
-}
+//package spring.session.demo.app
```
Comment (parveen-rx) [inline src/integration-test/groovy/spring/session/demo/app/SessionPluginSpec.groovy:1]: Shall we revert the commented spec and add/update test cases

**Review by parveen-rx — CHANGES_REQUESTED** [review]

File: src/integration-test/groovy/spring/session/demo/app/SessionPluginSpec.groovy (hunk 5 above)
Comment (RxL-Farhan-Ali) [reply src/integration-test/groovy/spring/session/demo/app/SessionPluginSpec.groovy:1]: Will do it in next enhancement added in TODO list

File: build.gradle (hunk 1 above)
Comment (RxL-Farhan-Ali) [reply build.gradle]: Removed

File: build.gradle (hunk 2 above)
Comment (RxL-Farhan-Ali) [reply build.gradle]: Updated.

File: grails-cli.yml (hunk 3 above)
Comment (RxL-Farhan-Ali) [reply grails-cli.yml]: updated.

File: settings.gradle (hunk 4 above)
Comment (RxL-Farhan-Ali) [reply settings.gradle]: Removed.

**Review by parveen-rx — CHANGES_REQUESTED** [review]
Please resolve code conflicts

@vsachinv : Please review the changes from your end

File: gradle.properties [hunk 6]
```diff
@@ -1,4 +1,7 @@
-#grailsVersion=3.1.1
-#gradleWrapperVersion=2.9
-grailsVersion=3.3.9
-gradleWrapperVersion=3.5
\ No newline at end of file
+grailsVersion=6.2.0
+grailsGradlePluginVersion=6.1.2
+version=7.0-JDK11-M1-local21
```
Comment (parveen-rx) [inline gradle.properties]: Add required version

**Review by parveen-rx — CHANGES_REQUESTED** [review]

File: gradle.properties (hunk 6 above)
Comment (RxL-Farhan-Ali) [reply gradle.properties]: yes let me update it.

**Review by parveen-rx — APPROVED** [review]

**Comment by parveen-rx** [comment]
@vsachinv : Please review and merge

CC: @RxL-Farhan-Ali

**Comment by parveen-rx** [comment]
@vsachinv : Please review when you get some time
Merging now

**Pull request merged**